src/redis_rdb.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <fstream>
+#include <string>
+
+#include "redis_db.h"
+
+namespace Redis {
+class RedisDatabase : public Database {
```

Comment:
@ColinChamber I think you can rename `RedisDatabase` to something like `DumpCreator` and move all the functionality from `Composer` to `DumpCreator`. Thus, you won't have `RedisDatabase` with just 1 method and inside it one private class and such long things like `RedisDatabase::Composer::MethodName()`. What do you think?

Reply:
Is the class name `DumpCreator` ok if the `parser` is added in the future?

Reply:
Do you mean the parser for importing the dump?

Reply:
Parse the Redis dump file. I know few people need it, but if someone wants to achieve it.

Reply:
So you think about the `RedisDatabase` class and its inner classes, for example, `DumpCreator` and `DumpLoader`/`DumpImporter`/etc. ?